### PR TITLE
Monomorphize interface methods and choices

### PR DESCRIPTION
**Pull request by akrmn**
DA-GHC PR: https://github.com/digital-asset/ghc/pull/132

Closes https://github.com/digital-asset/daml/issues/14842

### CODE REVIEW

**Review by remyhaemmerle-da — APPROVED** [review]
Thanks.

File: docs/source/daml/reference/interfaces.rst [hunk 1]
```diff
@@ -86,25 +86,24 @@ Interface Methods
...
       :end-before: -- INTERFACE_METHODS_TOP_LEVEL_END
 
-  - The ``Implements t I`` constraint means that the function can be applied to
-    any argument of type ``t`` such that:
-
-    - ``t`` is a template type and there exists an ``interface instance`` of
-      ``I`` for ``t``, **OR**
-    - ``t`` is the abstract interface type ``I`` itself.
+  - The first argument's type ``I`` means that the function can only be applied
+    to values of the interface type ``I`` itself. In particular, methods cannot
```
Comment (carrielaben-da) [inline docs/source/daml/reference/interfaces.rst:97]:
```suggestion
    to values of the interface type ``I`` itself. Methods cannot
```

File: docs/source/daml/reference/interfaces.rst [hunk 2]
```diff
@@ -142,6 +141,9 @@ Interface Choices
 - Interface choices work in a very similar way to template choices. Any contract
   of a template type for which an interface instance exists, will grant the
```
Comment (carrielaben-da) [inline docs/source/daml/reference/interfaces.rst:142]:
```suggestion
  of a template type for which an interface instance exists will grant the
```

**Review by carrielaben-da — COMMENTED** [review]
Just a couple of small things.

**Comment by akrmn** [comment]
> Just a couple of small things.

Thank you @carrielaben-da!

**Pull request merged**